common/credential-storage/src/backends/sqlite.rs

```diff
@@ -95,6 +95,20 @@ impl SqliteEcashTicketbookManager {
         Ok(())
     }
 
+    pub(crate) async fn contains_ticketbook_data(&self, data: &[u8]) -> Result<bool, sqlx::Error> {
+        let exists = sqlx::query(
+            r#"
+                SELECT * FROM ecash_ticketbook WHERE ticketbook_data = ?
```

Comment:
optimisation: rather than retrieving data, allocating it and then finally comparing it, why not do everything inside the db via `EXISTS` operator?

Reply:
Done